stream-chat-android-client/src/main/java/io/getstream/chat/android/client/ChatClient.kt

```diff
@@ -257,6 +257,8 @@ internal constructor(
     private val repositoryFactoryProvider: RepositoryFactory.Provider,
     @InternalStreamChatApi
     public val audioPlayer: AudioPlayer,
+    public val appName: String,
+    public val appVersion: String,
```

Comment:
We are adding more and more properties/methods on this class that don't apport functionality to our customers and make maintenance harder (This class is 4000+ LoC and isn't properly tested :disappointed: )

What about extracting it out of our `ChatClient`?
I think we can extract the `buildSdkTrackingHeaders()` method by building a lambda function within our `BaseChatModule` instance (We have all the needed data there, it needs to continue being a lambda because the `VERSION_PREFIX_HEADER.prefix` until the chat UI is initialized) and inject it on both classes that are using it (`HeadersInterceptor` and `SocketFactory`), both are created on this `BaseChatModule`